### PR TITLE
Syseng tweaks

### PR DESCRIPTION
**Pull request by goodboy**
various runtime machinery adjustments made as part of getting out multi-provider search engine going in #164.

@guilledk of particular note is the `_daemon.py` adjustments to use the new `tractor` context api to async spawn sub-daemons using a `pikerd` local `AsyncContextStack`.  I feel like this api style is starting to feel nice; interested to see what you think.

 Will be rebasing #168 onto this.

### CODE REVIEW

File: piker/_daemon.py [hunk 1]
```diff
@@ -44,10 +49,34 @@ class Services(BaseModel):
     actor_n: tractor._trionics.ActorNursery
     service_n: trio.Nursery
     debug_mode: bool  # tractor sub-actor debug mode flag
+    ctx_stack: AsyncExitStack
 
     class Config:
         arbitrary_types_allowed = True
 
+    async def open_remote_ctx(
```
Comment (goodboy) [inline piker/_daemon.py:57]: @guilledk this was the main bit i wanted your review on.

This let's us do deterministic, async setup/teardown from the root `pikerd` nicely.

File: piker/clearing/_ems.py [hunk 2]
```diff
@@ -179,31 +179,40 @@ async def execute_triggers(
...
-                        brid=None,
-
-                        symbol=sym,
-                        action=cmd['action'],
-                        price=submit_price,
-                        size=cmd['size'],
-                    )
+                    action = cmd['action']
+
+                    if action != 'alert':
```
Comment (goodboy) [inline piker/clearing/_ems.py:184]: This is just a bug fix, avoid sending limit orders when we set an alert.

File: piker/clearing/_ems.py [hunk 3]
```diff
@@ -642,35 +653,18 @@ async def _emsd_main(
...
-                )
+        # TODO: eventually support N-brokers
+        async with data.open_feed(
+            broker,
+            [symbol],
+            loglevel='info',
+        ) as feed:
+
+            # get a portal back to the client
+            async with tractor.wait_for_actor(client_actor_name) as portal:
```
Comment (goodboy) [inline piker/clearing/_ems.py:667]: This was a slight reorder to get the order mode to startup faster.

**Review by guilledk — APPROVED** [review]

File: piker/fsp/__init__.py [hunk 4]
```diff
@@ -62,6 +65,97 @@ async def latency(
             yield value
 
 
+async def fsp_compute(
```
Comment (goodboy) [inline piker/fsp/__init__.py:68]: This was just factored out of what was a closure to increase readability.

File: piker/data/feed.py [hunk 5]
```diff
@@ -108,7 +116,11 @@ def get_feed_bus(
     return _bus
 
 
-async def _setup_persistent_brokerd(brokername:  str) -> None:
+@tractor.context
```
Comment (goodboy) [inline piker/data/feed.py:119]: This was the main new bit: show's how we setup `brokerd` state using `tractor`'s new `Context` api.

**Pull request merged**